swap_meet/vendor.py

```diff
@@ -0,0 +1,94 @@
+class Vendor:
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []      
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        item_list = []
+
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+
+        return item_list
+
+    def swap_items(self, vendor, my_item, their_item):
+        contains_item = False
+
+        if my_item in self.inventory and their_item in vendor.inventory:
+            self.inventory.remove(my_item)
+            vendor.inventory.append(my_item)
+
+            vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            contains_item = True
+        else:
+            return contains_item
+        
+        return contains_item
+
+    def swap_first_item(self, vendor):
+        contains_item = False
+
+        if not self.inventory or not vendor.inventory:
+            return contains_item
+        else:
+            contains_item = self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
+
+        return contains_item
```

Comment:
This is really good. I do have a more compact version

```suggestion
        if  self.inventory and vendor.inventory:
            return self.swap_items(vendor, self.inventory[0], vendor.inventory[0])

        return False
```